includes/Core/Authentication/Clients/Google_Site_Kit_Client.php

```diff
@@ -10,89 +10,99 @@
+class Google_Site_Kit_Client extends Google_Client {
 
 	/**
 	 * Callback to pass a potential exception to while refreshing an access token.
 	 *
 	 * @since n.e.x.t
 	 * @var callable|null
 	 */
-	protected $token_exception_callback = null;
+	protected $token_exception_callback;
 
 	/**
-	 * Construct the Google Client.
+	 * Construct the Google client.
 	 *
-	 * @since 1.1.2
+	 * @since n.e.x.t
 	 *
-	 * @param array $config Proxy client configuration.
+	 * @param array $config Client configuration.
 	 */
 	public function __construct( array $config = array() ) {
-		if ( ! empty( $config['proxy_base_path'] ) ) {
-			$this->setProxyBasePath( $config['proxy_base_path'] );
-		}
 		if ( isset( $config['token_exception_callback'] ) ) {
 			$this->setTokenExceptionCallback( $config['token_exception_callback'] );
 		}
 
-		unset( $config['proxy_base_path'], $config['token_exception_callback'] );
+		unset( $config['token_exception_callback'] );
 
 		parent::__construct( $config );
 	}
 
 	/**
-	 * Sets the base URL to the proxy.
+	 * Sets the function to be called when fetching an access token results in an exception.
 	 *
 	 * @since n.e.x.t
 	 *
-	 * @param string $base_path Proxy base URL.
+	 * @param callable $exception_callback Function accepting an exception as single parameter.
 	 */
-	public function setProxyBasePath( $base_path ) {
-		$this->proxy_base_path = untrailingslashit( $base_path );
+	public function setTokenExceptionCallback( callable $exception_callback ) {
+		$this->token_exception_callback = $exception_callback;
 	}
 
 	/**
-	 * Sets the function to be called when fetching an access token results in an exception.
+	 * Adds auth listeners to the HTTP client based on the credentials set in the Google API Client object.
 	 *
 	 * @since n.e.x.t
 	 *
-	 * @param callable $exception_callback Function accepting an exception as single parameter.
+	 * @param ClientInterface $http The HTTP client object.
+	 * @return ClientInterface The HTTP client object
 	 */
-	public function setTokenExceptionCallback( callable $exception_callback ) {
-		$this->token_exception_callback = $exception_callback;
+	public function authorize( ClientInterface $http = null ) {
+		if ( $this->isUsingApplicationDefaultCredentials() ) {
+			return parent::authorize( $http );
+		}
+
+		$token = $this->getAccessToken();
+		if ( isset( $token['refresh_token'] ) && $this->isAccessTokenExpired() ) {
+			$callback = $this->getConfig( 'token_callback' );
+
+			try {
+				$creds = $this->fetchAccessTokenWithRefreshToken( $token['refresh_token'] );
+				if ( $callback ) {
+					// Due to original callback signature this can only accept the token itself.
+					call_user_func( $callback, '', $creds['access_token'] );
+				}
+			} catch ( Exception $e ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement
+				// Pass exception to special callback if provided.
+				if ( $this->token_exception_callback ) {
+					call_user_func( $this->token_exception_callback, $e );
+				}
```

Comment:
Good catch! I think this should be re-thrown regardless though, since that is the behavior of the default client, nothing is caught there. We would catch that exception in our own code then.